app/models/server/models/Invites.js

```diff
@@ -0,0 +1,46 @@
+import { Base } from './_Base';
+
+class Invites extends Base {
+	constructor() {
+		super('invites');
+
+		this.tryEnsureIndex({ hash: 1 });
+	}
+
+	findOneByHash(hash) {
+		return this.findOne({ hash });
+	}
+
+	findOneByUserRoomMaxUsesAndExpiration(userId, rid, maxUses, daysToExpire) {
+		const query = {
+			rid,
+			userId,
+			days: daysToExpire,
+			maxUses,
+		};
+
+		if (daysToExpire > 0) {
+			query.expires = {
+				$gt: Date.now(),
+			};
+		}
+
+		if (maxUses > 0) {
+			query.uses = 0;
```

Comment:
AFAIK we should return the token if there are usages left. Right @sampaiodiego ?

Reply:
yes.. even if there are only 1 usage left, we can return that same link/token. this is to reduce the amount of spare links generated, and if they end up reaching usage limit, a new one can be generated =)